airflow/providers/microsoft/mssql/hooks/mssql.py

```diff
@@ -51,6 +51,13 @@ def get_conn(
         )
         return conn
 
+    def get_uri(self) -> str:
+        from urllib.parse import urlsplit, urlunsplit
+
+        r = list(urlsplit(super().get_uri()))
+        r[0] = "mssql+pymssql"  # fix to use pymssql driver
+        return urlunsplit(r)
```

Comment:
Is this the right way?
This is what we have in OdbcHook
https://github.com/apache/airflow/blob/46bbfdade0638cb8a5d187e47034b84e68ddf762/airflow/providers/odbc/hooks/odbc.py#L34

and users can override this
https://airflow.apache.org/docs/apache-airflow-providers-odbc/stable/connections/odbc.html#configuring-the-connection

Reply:
You are right, we could add a `DEFAULT_SQLALCHEMY_SCHEME = 'mssql+pymssql'` to the `MsSqlHook` class as well as a constructor parameter and use it.

Reply:
@eladkal Done ;)

Reply:
By the way the docstring in odbc provider for `sqlalchemy_scheme` is copy pasted: 
https://github.com/apache/airflow/blob/46bbfdade0638cb8a5d187e47034b84e68ddf762/airflow/providers/odbc/hooks/odbc.py#L84

Reply:
Still some pre-commits failed. I recommend (as in CONTRIBUTORS_GUIDE) to install pre-commit - it will fix most things automatically. Also `breeze static-check --last-commit` re-runs all relevant pre-commits for you automatically on last commit (so you need to rebase&squash to single commit before).

Reply:
> By the way the docstring in odbc provider for sqlalchemy_scheme is copy pasted:

Yep. Fixing it is a good idea - you can even do it in this PR or open another PR.

Reply:
> > By the way the docstring in odbc provider for sqlalchemy_scheme is copy pasted:
> 
> Yep. Fixing it is a good idea - you can even do it in this PR or open another PR.

I made another PR for odbc docstring fix: #25421 :wink:

Reply:
> Still some pre-commits failed. I recommend (as in CONTRIBUTORS_GUIDE) to install pre-commit - it will fix most things automatically. Also `breeze static-check --last-commit` re-runs all relevant pre-commits for you automatically on last commit (so you need to rebase&squash to single commit before).

Should be good now :smiley: